src/main/java/edu/upf/taln/dri/lib/model/util/ObjectGenerator.java

```diff
@@ -1600,307 +1599,17 @@ else if(titleLang.trim().toLowerCase().equals("es")) {
-										if(emailOpt.isPresent() && emailOpt.get() != null && !emailOpt.get().equals("")) {
-											newAuthor.setEmail(normalizeText(emailOpt.get()).trim());
-										}
-									}
-									
-									// Affiliation
-									List<Annotation> affiliationList = GateUtil.getAnnInDocOrderContainedAnn(cacheManager.getGateDoc(), ImporterGROBID.GROBIDannSet, "affiliation", authorAnn);
-									if(affiliationList != null && affiliationList.size() > 0) {
-										for(Annotation affil : affiliationList) {
-											if(affil != null) {
-												InstitutionImpl newAffiliation = new InstitutionImpl(cacheManager);
-												
-												// Affiliation fields:
-												// ADDED: orgName, address, country
-												// TO ADD: marker, URL, labs, instits. addr, region, settlement, acronym 
-												
-												// Name
-												List<Annotation> orgNameList = GateUtil.getAnnInDocOrderContainedAnn(cacheManager.getGateDoc(), ImporterGROBID.GROBIDannSet, "orgName", affil);
-												if(orgNameList != null && orgNameList.size() > 0) {
-													Optional<String> orgNameOpt = GateUtil.getAnnotationText(orgNameList.get(0), cacheManager.getGateDoc());
-													if(orgNameOpt.isPresent() && orgNameOpt.get() != null && !orgNameOpt.get().equals("")) {
-														newAffiliation.setName(normalizeText(orgNameOpt.get()).trim());
-													}
-												}
-												
-												// Address
-												List<Annotation> addressList = GateUtil.getAnnInDocOrderContainedAnn(cacheManager.getGateDoc(), ImporterGROBID.GROBIDannSet, "address", affil);
-												if(addressList != null && addressList.size() > 0) {
-													Optional<String> addressOpt = GateUtil.getAnnotationText(addressList.get(0), cacheManager.getGateDoc());
-													if(addressOpt.isPresent() && addressOpt.get() != null && !addressOpt.get().equals("")) {
-														newAffiliation.setAddress(normalizeText(addressOpt.get()).trim());
-													}
-												}
-												
-												// Country
-												List<Annotation> countryList = GateUtil.getAnnInDocOrderContainedAnn(cacheManager.getGateDoc(), ImporterGROBID.GROBIDannSet, "country", affil);
-												if(countryList != null && countryList.size() > 0) {
-													Optional<String> countryOpt = GateUtil.getAnnotationText(countryList.get(0), cacheManager.getGateDoc());
-													if(countryOpt.isPresent() && countryOpt.get() != null && !countryOpt.get().equals("")) {
-														newAffiliation.setState(normalizeText(countryOpt.get()).trim());
-													}
-												}
-												
-												if(newAffiliation.getName() != null && !newAffiliation.getName().equals("")) {
-													newAuthor.addAffiliation(newAffiliation);
-												}
-												
-											}
-										}
-									}
-									
-									resultImpl.addAuthor(newAuthor);
-								}
-							}
-						}
-					}
-					
+				try {
+					SourceENUM sourceType = SourceENUM.valueOf(sourceDocFeature);
```

Comment:
The new code. This is surrounded in a `try-catch`
This code should:
- Turn `sourceDocFeature` into an enum or fail and continue without author extraction.
- Pass this enum into `ExtractorFactory`
- `ExtractorFactory` picks the appropriate extractor (GROBID, JATS, or PDFEXT) or throws an exception if its none of the three (failing, and continuing without author extraction)
- The selected `Extractor` is (mostly) the same code that was removed from `ObjectGenerator` based on the particular if-else block.
- The selected `Extractor` extracts the authors, filling a list rather than adding them to `resultImpl` as it goes along.
- The results of the `Extractor` get added into `resultImpl` all at once using the new method we introduced.